gensim/similarities/index.py

```diff
@@ -87,3 +87,77 @@ def most_similar(self, vector, num_neighbors):
             vector, num_neighbors, include_distances=True)
 
         return [(self.labels[ids[i]], 1 - distances[i] / 2) for i in range(len(ids))]
+
+try:
```

Comment:
Create new file for your implementation, because user can't import FaissIndexer if annoy is'nt installed (and v.v.)